lib/batch_reporter.ex

```diff
@@ -0,0 +1,82 @@
+defmodule TelemetryInfluxDB.BatchReporter do
+  @moduledoc """
+  This module handles batching events to be reported to Influx. It will call the provided
+  report function with any events that have been enqueued since the last report. If a batch_size
+  is not provided, the batch size will default to 1, which will send all events separately.
+  """
+
+  use GenServer
+
+  defmodule State do
+    defstruct [:report_fn, :batch_size, report_scheduled?: false, unreported_events: []]
+
+    def enqueue_event(state, event) do
+      %{state | unreported_events: state.unreported_events ++ [event]}
```

Comment:
#optional This would be more work in other parts of the code, but given how lists are implemented in Elixir, would it be more efficient to prepend events here and then reverse just before sending a batch?

The extra code complexity may not be worth the performance gain, but it probably depends on how many events will get queued up between batch sends.